.github/workflows/lint.yml

```diff
@@ -20,18 +20,41 @@ jobs:
         uses: actions/setup-python@v4
         with:
           python-version: 3.8
+      - name: Setup flake8 annotations
+        uses: rbialon/flake8-annotations@v1
       - name: Install linter
         run: |
           pip install flake8 pylint
-      - name: Lint
+      - name: Lint codebase
         run: |
           make lint
       - name: Install bibtex linter
         run: |
           pip install bibtexparser
       - name: Check bibtex
         run: |
-          for bibfile in docs/source/_static/*.bib; do
-              echo "Checking $(basename $bibfile) for validation errors"
-              python3 scripts/firedrake-preprocess-bibtex --validate $bibfile
+          cd docs
+          make validate-bibtex
+      - name: Install YAML linter
+        run: |
+          pip install yamllint
+      - name: Lint YAML files
+        run: |
+          yamllint -f github .github/workflows/*.yml
```

Comment:
You could consider https://github.com/rhysd/actionlint which knows about the schema of github actions yaml files instead.